### PR TITLE
feat: Uppercase/Lowercase All Non-literal Tokens With Formatting

### LINKED ISSUE #1174: Capitalize All Non-literal Tokens In The Document With Formatting
**Is your feature request related to a problem?**

The general style guide for COBOL says that all the non-literal tokens (i.e. everything that is not inside quotes) should be in the upper case.

**Describe the solution you'd like**

Add capitalization as a part of the existing formatting feature. Add a switch to the settings to enable or disable it.

### PR DESCRIPTION
**Pull request by Nurkambay**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/package.json [hunk 1]
```diff
@@ -523,6 +523,16 @@
...
+        },
+        "cobol-lsp.formatting": {
+          "type": "string",
+          "default": "Uppercase",
+          "enum": [
+            "Uppercase",
+            "Lowercase",
+            "None"
+          ],
+          "description": "Formatting settings"
```
Comment (Nurkambay) [inline clients/cobol-lsp-vscode-extension/package.json:535]: @zeibura can you please check this naming and description?

**Comment by ishche** [comment]
Since program id has not been changed, should we preserve program calls?

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/formations/CapitalFormation.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    return context.getResult();
+  }
+
+  private void processLine(StringBuilder builder, CobolDocumentModel.Line line, Context context) {
+    int maxLen = Math.min(line.getText().length(), MAX_LEN);
+    context.startNewLine();
+
+    // Check for commented line
+    if (maxLen >= MIN_LEN) {
+      if (line.getText().charAt(MIN_LEN - 1) == '*') {
```
Comment (ishche) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/formations/CapitalFormation.java:103]: Should in-line comments be supported too?

Comment (Nurkambay) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/formations/CapitalFormation.java:103]: Good point

File: clients/cobol-lsp-vscode-extension/package.json [hunk 3]
```diff
@@ -523,6 +523,16 @@
             "NATIVE - Use platform binary"
           ],
           "description": "COBOL LS server runtime."
+        },
+        "cobol-lsp.formatting": {
+          "type": "string",
+          "default": "Uppercase",
```
Comment (VitGottwald) [inline clients/cobol-lsp-vscode-extension/package.json]: Default should be "None" for backward compatibility.

Comment (Nurkambay) [reply clients/cobol-lsp-vscode-extension/package.json]: Make sense, fixed

**Review by zeibura — APPROVED** [review]

File: clients/cobol-lsp-vscode-extension/package.json (hunk 1 above)
Comment (zeibura) [reply clients/cobol-lsp-vscode-extension/package.json:535]: What does it do?

**Comment by Nurkambay** [comment]
> Since program id has not been changed, should we preserve program calls?

What do you mean?

**Comment by ishche** [comment]
> Since program id has not been changed, should we preserve program calls?

Seems that it's not the case, because cobol expects program names as literals.

**Review by ishche — APPROVED** [review]

**Pull request merged**